geoalchemy2/functions.py

```diff
@@ -92,6 +94,14 @@ def __init__(self, *args, **kwargs):
         expr = kwargs.pop('expr', None)
         if expr is not None:
             args = (expr,) + args
+
+        args = [
+            bindparam(elem.name, unique=True, value=elem,
+                      type_=types._GISType(from_text=elem.name, name=elem.name))
+            if isinstance(elem, elements.HasFunction) else elem
+            for elem in args
+        ]
```

Comment:
I wonder why this is not simply:

```py
    args = [elem.function_expr if isintance(elem, elements.HasFunction) else elem for elem in args]
```

?

Reply:
I think it could be that also.  but the way we have SQL concepts sprinkled around in multiple places is part of what makes this hard to follow.